homeassistant/components/evohome/entity.py

```diff
@@ -68,37 +62,33 @@ async def async_zone_svc_request(self, service: str, data: dict[str, Any]) -> No
         """Process a service request (setpoint override) for a zone."""
         raise NotImplementedError
 
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
-        """Return the evohome-specific state attributes."""
-        status = self._device_state_attrs
-        if SZ_SYSTEM_MODE_STATUS in status:
-            convert_until(status[SZ_SYSTEM_MODE_STATUS], SZ_TIME_UNTIL)
-        if SZ_SETPOINT_STATUS in status:
-            convert_until(status[SZ_SETPOINT_STATUS], SZ_UNTIL)
-        if SZ_STATE_STATUS in status:
-            convert_until(status[SZ_STATE_STATUS], SZ_UNTIL)
-
-        return {"status": convert_dict(status)}
-
     async def async_added_to_hass(self) -> None:
         """Run when entity about to be added to hass."""
-        async_dispatcher_connect(self.hass, DOMAIN, self.async_refresh)
+        await super().async_added_to_hass()
 
+        async_dispatcher_connect(self.hass, DOMAIN, self.process_signal)
 
-class EvoChild(EvoDevice):
+    async def async_update(self) -> None:
+        """Get the latest state data."""
+        await super().async_update()
+
+        self._attr_extra_state_attributes = {self._evo_id_attr: self._evo_device.id}
```

Comment:
Why do we do this?

Reply:
This is the `system_id`, `zone_id`, `dhw_id` attrs we spoke about before. 

This attr is fixed for the life of the entity, but this dict is built up by the subclasses with data that does vary, such at the current/next setpoints.

Reply:
No but I mean, why do we do this in `async_update`?

Reply:
`_attr_extra_state_attributes` returns a dict with useful data that is not part of the Climate/WaterHeater schema, such as setpoints schedules, faults, etc...

My thinking is when the **DataUpdateCoordinator.async_refresh()** is called, it:
 - first calls the client library's `update()` method, and only _then_ can these values change
 - then calls the entities' `async_update()` method, where I rebuild the dict used to make  `_attr_extra_state_attributes`

Reply:
But async_refresh does not call update or async update?

Reply:
Oh - I've got it the wrong way around! `CoordinatorEntity.async_update` invokes `DataUpdateCoordinator.async_request_refresh()`!

Reply:
So, what is the HA way?